src/main/java/com/google/jenkins/plugins/k8sengine/CredentialsUtil.java

```diff
@@ -0,0 +1,120 @@
+import jenkins.model.Jenkins;
+
+/** Provides a library of utility functions for credentials-related work. */
+public class CredentialsUtil {
+  /**
+   * Get the Google Robot Credentials for the given credentialsId.
+   *
+   * @param itemGroup A handle to the Jenkins instance. Must be non-null.
+   * @param domainRequirements A list of domain requirements. Must be non-null.
+   * @param credentialsId The ID of the GoogleRobotCredentials to be retrieved from Jenkins and
+   *     utilized for authorization. Must be non-empty or non-null and exist in credentials store.
+   * @return Google Robot Credential for the given credentialsId.
+   * @throws AbortException If there was an issue retrieving the Google Robot Credentials.
+   */
+  public static GoogleRobotCredentials getRobotCredentials(
+      ItemGroup itemGroup,
+      ImmutableList<DomainRequirement> domainRequirements,
+      String credentialsId)
+      throws AbortException {
+    Preconditions.checkNotNull(itemGroup);
+    Preconditions.checkNotNull(domainRequirements);
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(credentialsId));
+
+    GoogleRobotCredentials robotCreds =
+        CredentialsMatchers.firstOrNull(
+            CredentialsProvider.lookupCredentials(
+                GoogleRobotCredentials.class, itemGroup, ACL.SYSTEM, domainRequirements),
+            CredentialsMatchers.withId(credentialsId));
+
+    if (robotCreds == null) {
+      throw new AbortException(Messages.ClientFactory_FailedToRetrieveCredentials(credentialsId));
+    }
+
+    return robotCreds;
+  }
+
+  /**
+   * Get the Credential from the Google robot credentials for GKE access.
+   *
+   * @param robotCreds Google Robot Credential for desired service account.
+   * @return Google Credential for the service account.
+   * @throws AbortException if there was an error initializing HTTP transport.
+   */
+  public static Credential getGoogleCredential(GoogleRobotCredentials robotCreds)
+      throws AbortException {
+    Credential credential;
+    try {
+      credential = robotCreds.getGoogleCredential(new ContainerScopeRequirement());
+    } catch (GeneralSecurityException gse) {
+      throw new AbortException(
+          Messages.ClientFactory_FailedToInitializeHTTPTransport(gse.getMessage()));
+    }
+
+    return credential;
+  }
+
+  /**
+   * Wrapper to get access token for service account with this credentialsId.
+   *
+   * @param credentialsId The service account credential's id. Must be non-null.
+   * @return Access token from OAuth to allow kubectl to interact with the cluster.
+   * @throws java.io.IOException If an error occurred fetching the access token.
+   */
+  static String getAccessToken(String credentialsId) throws IOException {
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(credentialsId));
+    GoogleRobotCredentials robotCreds =
+        getRobotCredentials(Jenkins.get(), ImmutableList.<DomainRequirement>of(), credentialsId);
```

Comment:
Potentially overkill but may be good to let the user pass in the Jenkins context, and that should be used in `doCheckCredentialsId` that I mentioned above.

I would convert this block into a separate method which takes and uses the Jenkins object, and in the method body for the existing method header, you can just call the new method with `Jenkins.get()` as the parameter, so that existing calls don't change but this new use case is supported.

Reply:
Done.

Reply:
I just noticed but this should be itemGroup instead of Jenkins.get()